pkg/storage/sql/sql_test.go

```diff
@@ -0,0 +1,301 @@
+package sql
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/openware/ika"
+	"github.com/openware/kaigara/pkg/aes"
+	"github.com/openware/kaigara/pkg/plaintext"
+	"github.com/openware/kaigara/pkg/transit"
+	"github.com/openware/kaigara/types"
+	"github.com/openware/pkg/database"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type Suite struct {
+	suite.Suite
+	deploymentID string
+	appNames     []string
+	scopes       []string
+	configs      map[string]database.Config
+	encryptors   map[string]map[string]types.Encryptor
+}
+
+func (s *Suite) SetupSuite() {
+	vaultAddr := os.Getenv("KAIGARA_VAULT_ADDR")
+	vaultToken := os.Getenv("KAIGARA_VAULT_TOKEN")
+
+	s.deploymentID = "opendax_uat"
+	s.appNames = []string{"barong", "finex", "peatio"}
+	s.scopes = []string{"private", "public", "secret"}
+
+	path, err := os.Getwd()
+	if err != nil {
+		panic(err)
+	}
+
+	type Config struct {
+		Name     string          `yaml:"name"`
+		DbConfig database.Config `yaml:"database"`
+	}
+	cfgs := make(map[string]Config)
+	ika.ReadConfig(path+"/config.yml", &cfgs)
+	s.configs = make(map[string]database.Config)
+	for _, cfg := range cfgs {
+		s.configs[cfg.Name] = cfg.DbConfig
+	}
+	aesEncrypt, err := aes.NewAESEncryptor([]byte("1234567890123456"))
+	if err != nil {
+		panic(err)
+	}
+
+	plainEncrypt := plaintext.NewPlaintextEncryptor([]byte(""))
+	s.encryptors = map[string]map[string]types.Encryptor{
+		"aes": {
+			"barong": aesEncrypt,
+			"finex":  aesEncrypt,
+			"peatio": aesEncrypt,
+		},
+		"transit": {
+			"barong": transit.NewVaultEncryptor(vaultAddr, vaultToken, "barong"),
+			"finex":  transit.NewVaultEncryptor(vaultAddr, vaultToken, "finex"),
+			"peatio": transit.NewVaultEncryptor(vaultAddr, vaultToken, "peatio"),
+		},
+		"plaintext": {
+			"barong": plainEncrypt,
+			"finex":  plainEncrypt,
+			"peatio": plainEncrypt,
+		},
+	}
+}
+
+func TestInit(t *testing.T) {
+	suite.Run(t, new(Suite))
+}
+
+func getEntries(ss *StorageService, appName, scope string) map[string]interface{} {
+	err := ss.Read(appName, scope)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("Loading ss for %s\n", scope)
+	data, err := ss.GetEntries(appName, scope)
+	if err != nil {
+		panic(err)
+	}
+	return data
+}
+
+func getEntry(ss *StorageService, appName, scope, name string) interface{} {
+	err := ss.Read(appName, scope)
+	if err != nil {
+		panic(err)
+	}
+
+	entry, err := ss.GetEntry(appName, scope, name)
+	fmt.Printf("entry: %s\n", entry)
+	if err != nil {
+		panic(err)
```

Comment:
Would be nice to call these methods reloadEntries or getEntriesReload to signify that we re-read the scope every time